src/udp_wrap.cc

```diff
@@ -91,7 +130,7 @@ void UDPWrap::Initialize(Local<Object> target,
   Environment* env = Environment::GetCurrent(context);
 
   Local<FunctionTemplate> t = env->NewFunctionTemplate(New);
-  t->InstanceTemplate()->SetInternalFieldCount(1);
+  t->InstanceTemplate()->SetInternalFieldCount(kUDPWrapBaseField + 1);
```

Comment:
This is error prone. Things will break badly when someone changes `kUDPWrapBaseField` from 1 to 0. Can I suggest this alternative?
```c++
class BaseObject {
 public:
  enum InternalFields { kSlot, kCount };
  // ...
};

class AsyncWrap : public BaseObject {
 public:
  enum InternalFields { kCount = BaseObject::InternalFields::kCount };
  // ...
};

// propagating upwards until...

class UDPWrapBase : public HandleWrap {
 public:
  enum InternalFields { kSlot = HandleWrap::InternalFields::kCount, kCount };
  // ...
};

class UDPWrap : public UDPWrapBase {
 public:
  enum InternalFields { kCount = UDPWrapBase::InternalFields::kCount };
  // ...
};

void UDPWrap::Initialize(...) {
  // ...
  t->InstanceTemplate()->SetInternalFieldCount(InternalFields::kCount);
  // ...
}
```

Reply:
> This is error prone. Things will break badly when someone changes `kUDPWrapBaseField` from 1 to 0. 

Which is why I guess nobody would make that change? ;)

Your suggestion sounds fine but I wouldn’t consider it necessary for this PR…

Reply:
For now, I'll add a code comment warning folks away from that change along with a `CHECK`

Reply:
That's not good enough. If for whatever reason BaseObject needs to use two slots instead of one, the code here will break in unexpected and probably interesting ways. There's no reason not to get this right from the start.

Reply:
It appears that there are nearly 60 places in `src/*` where similar break could occur. Your suggested fix is good but likely should be made throughout. I'll open a separate PR for it.

Reply:
I mean, we already have this problem with StreamBase/StreamReq. I’m personally good with a separate PR.

Reply:
Other PR: https://github.com/nodejs/node/pull/31960

If that one lands first, I'll update this one, if this one lands first, I'll update that one